### PR TITLE
Sequential serial portion

### PR DESCRIPTION
**Pull request by pde**
Implement and test SerialSeq, an optional portion of the Serial Number the client can request
    
  - Intended to be used for clients that want a portion of the Serial Number to
      be sequential for auditing purposes.  Let's Encrypt will use this:
    
    https://github.com/letsencrypt/boulder/issues/84

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -35,11 +35,12 @@ type SigningProfile struct {
...
-	Expiry       time.Duration
-	Backdate     time.Duration
-	Provider     auth.Provider
-	RemoteServer string
+	Policies        []asn1.ObjectIdentifier
+	Expiry          time.Duration
+	Backdate        time.Duration
+	Provider        auth.Provider
+	RemoteServer    string
+	UseSerialSeq  bool
```
Comment (lziest) [inline config/config.go]: 'go fmt' is required.

File: api/sign/sign.go [hunk 2]
```diff
@@ -78,12 +78,13 @@ func NewHandlerFromSigner(signer signer.Signer) (h *api.HTTPHandler, err error)
...
-	Subject  *signer.Subject `json:"subject,omitempty"`
-	Profile  string          `json:"profile"`
-	Label    string          `json:"label"`
+	Hostname  string          `json:"hostname"`
+	Hosts     []string        `json:"hosts"`
+	Request   string          `json:"certificate_request"`
+	Subject   *signer.Subject `json:"subject,omitempty"`
+	Profile   string          `json:"profile"`
+	Label     string          `json:"label"`
+	SerialSeq string          `json:"serial_sequence"`
```
Comment (kisom) [inline api/sign/sign.go]: This should probably have omitempty in the struct tags.

**Comment by pde** [comment]
All test & lint issues should now be resolved, and reviewers' comments have been addressed, but travis is still failing because of a goveralls token issue that I doubt I can fix.

**Comment by kisom** [comment]
LGTM.

**Comment by kisom** [comment]
Test failure is due to well-known coveralls failure. All Travis tests pass.

**Pull request merged**